### DISCUSSION TITLE
Is it possible to have read/write permissions and restrictions in CUE?

### DISCUSSION THREAD

**Discussion by singhsuryansh12** [cue-lang/cue#1923] [Q&A]
:wave: Hello, team!
I’m wondering if it is possible set read/write permissions or restrictions in CUE for different files, packages or definitions?
For eg: Let’s say I have written multiple packages under cue.mod/pkg/domain.com and I have imported them under the outer cueConfig.cue file. The packages were written by separate teams and they contain separate definitions. Does CUE provide me the feature to limit the access of one team’s package, such that any other team’s member won’t be able to make a change to the definitions or the config, and doing so will result in an error?

**Comment by mvdan** [comment]
CUE packages are just plain files and directories on disk. How could it possibly restrict who edits those files on disk? Perhaps I'm misunderstanding your question.

Typically, if you have a company with employees and you want to restrict which employees can change parts of your codebase, you would do it at a VCS level. For example, split your modules or packages across multiple git repositories, and only give write access to each git repository to the set of people you want.
